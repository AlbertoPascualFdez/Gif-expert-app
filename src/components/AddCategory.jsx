import { useState } from "react"


export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState("");

    const onInputChange = (e) =>{
        //console.log(e.target.value)
        setInputValue(e.target.value);
    }

    const onSubmit = (e) =>{
        e.preventDefault();
        const trimedValue = inputValue.trim();

        console.log(inputValue)



        if(trimedValue.length <= 1) return;

        //setCategories(cat => [inputValue, ...cat])
        onNewCategory(trimedValue);

        setInputValue("");
    }
    

    return (
        <form onSubmit={ onSubmit}>
            <input type="text" placeholder="Buscar gifs" value={inputValue} onChange={onInputChange}/>
        </form>
        
    )
}
