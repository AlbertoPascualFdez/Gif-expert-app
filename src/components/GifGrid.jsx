import { GifItem } from "./GifItem";
import { useFetchHook } from "../hooks/useFetchHook";


export const GifGrid = ({category}) => {

//    const [images, setImages]  = useState([])

    const {images, isLoading} = useFetchHook(category);

    //no se debe poner la ejecucion de una funcion directamente en un functional component, ya que se ejecutara cada vez que se re renderice, y puede que no queramos eso.
    //getGifs(category);

//    const getImages = async() =>{
//        const newImages = await getGifs(category);
//        console.log("hola", newImages);
//        setImages(newImages);
//    }

    //se ejecuta en determinadas situaciones (cuando cambia el valor de una variable, en el primer renderizado, etc)
//    useEffect(() =>{
            //opcion 1
           // getGifs(category).then(newImages => setimages(newImages));

           //opcion 2
//           getImages();
//        },
//        [] //dependencias vacias -> se ejecuta solo una vez
//    )

    return (
        <>
            <h3>{category}</h3>

            {
                //isLoading ? (<h2>Cargando...</h2>): null
                isLoading && (<h2>Cargando...</h2>)
            }
            

            <div className="card-grid">
                {
                images.map((image) => (
                //images.map(({id, title}) => (
                   /*  <li key={id}>{title}</li> */
                   <GifItem key={image.id} {...image}/>//manda todas las propiedades que tenga el objeto image
                ))
                }
            </div>
 
        </>
    )
}
