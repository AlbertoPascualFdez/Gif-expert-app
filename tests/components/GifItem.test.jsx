import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('<GifItem>', () => {

    const title = "Saitama";
    const url = "https://one-punch/"
    
    test('Match con el snapshot', () => {
        const {container} = render(<GifItem title={title} url={url} />);

        expect(container).toMatchSnapshot();
    })

    test('Url y alt indicado', () => {
        render(<GifItem title={title} url={url} />);
        
        const {src, alt} = screen.getByRole("img");
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Title aparece', () => {
        render(<GifItem title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy();
    })
    
    
    
})