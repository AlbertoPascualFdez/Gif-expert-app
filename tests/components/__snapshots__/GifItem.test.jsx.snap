// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem> Match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
